### PR TITLE
Adding more refresh logging and converting code to Coroutines

### PR DESCRIPTION
**Pull request by geekygecko**
## Description

This change started by adding extra logging to help debug slow refreshing. I was finding it hard to add these timing logs so I converted some of the code from RxJava to Coroutines. The main method converted is `PodcastManager.refreshPodcastInBackground`. This called quite a few blocking methods so I have added alternative methods with the suffix `Suspend` until we can remove them. 

## Testing Instructions

I have added a new test for the rewrite but maybe just opening the podcast page and also running a refresh to see the new timings appear.

```
BgTask: Refresh - sync update - 385 ms
BgTask: Refresh - process server response - 1 ms
```

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [x] I have considered whether it makes sense to add tests for my changes
- [x] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml`
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [x] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/sync/PodcastRefresher.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+class PodcastRefresher @Inject constructor(
+    private val episodeManager: EpisodeManager,
+    private val appDatabase: AppDatabase,
+    private val cacheServiceManager: PodcastCacheServiceManager,
+    private val crashLogging: CrashLogging,
+) {
+    suspend fun refreshPodcast(existingPodcast: Podcast, playbackManager: PlaybackManager) {
+        try {
+            val podcastResponse = cacheServiceManager.getPodcastResponse(existingPodcast.uuid)
+            if (podcastResponse.wasCached() || podcastResponse.notModified()) {
```
Comment (geekygecko) [inline modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/sync/PodcastRefresher.kt:31]: With my tests, the `wasCached` method never thought this call was cached. If the app already has the correct version, the server will return a 304 status code. I have added this as an extra check.

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/PodcastViewModel.kt [hunk 2]
```diff
@@ -210,7 +212,10 @@ class PodcastViewModel
     }
 
     fun updatePodcast(existingPodcast: Podcast) {
-        podcastManager.refreshPodcastInBackground(existingPodcast, playbackManager)
+        // Refresh the podcast application coroutine scope so the podcast continues to update if the view model is closed
+        applicationScope.launch {
+            podcastManager.refreshPodcast(existingPodcast, playbackManager)
+        }
```
Comment (geekygecko) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/PodcastViewModel.kt:218]: I thought about moving this to a work manager task but as ideally, the podcast would be refreshed instantly I have left it as a Coroutine launch.

**Review by mebarbosa — APPROVED** [review]
LGTM

**Pull request merged**